coast/gridded.py

```diff
@@ -0,0 +1,865 @@
+        will vary in time (and space). This function will compute the vertical
+        scale factors e3t, e3u, e3v, e3f and e3w by using the sea surface height
+        field (ssh variable) and initial scale factors from the domain_cfg file.
+        The vertical scale factors will be computed at the same model time as the 
+        ssh and if the ssh field is averaged in time then the scale factors will
+        also be time averages. 
+        
+        A t-grid NEMO object containing the ssh variable must be passed in. Either 
+        the domain_cfg path must have been passed in as an argument when the NEMO 
+        object was created or it must be passed in here using the dom_fn argument.
+        
+        e.g. e3t,e3v,e3f = coast.NEMO.get_e3_from_ssh(nemo_t,true,false,true,true,false)
+        
+        Parameters
+        ----------
+        nemo_t : (COAsT.NEMO), NEMO object on the t-grid containing the ssh variable
+        e3t : (boolean), true if e3t is to be returned. Default True.
+        e3u : (boolean), true if e3u is to be returned. Default False.
+        e3v : (boolean), true if e3v is to be returned. Default False.
+        e3f : (boolean), true if e3f is to be returned. Default False.
+        e3w : (boolean), true if e3w is to be returned. Default False.
+        dom_fn : (str), Optional, path to domain_cfg file. 
+
+        Returns
+        -------
+        Tuple of xarray.DataArrays
+        (e3t, e3u, e3v, e3f, e3w)
+        Only those requested will be returned, but the ordering is always the same.
+
+        '''
+        e3_return = []
+        try:
+            ssh = nemo_t.dataset.ssh
+        except AttributeError:
+            print('The nemo_t dataset must contain the ssh variable.')
+            return
+        if 't_dim' not in ssh.dims:
+            ssh = ssh.expand_dims('t_dim', axis=0)
+
+        # Load domain_cfg
+        if dom_fn is None:
+            dom_fn = nemo_t.filename_domain
+        try:
+            ds_dom = xr.open_dataset(dom_fn).squeeze().rename(
+                {'z': 'z_dim', 'x': 'x_dim', 'y': 'y_dim'})
+        except OSError:
+            print(f'Problem opening domain_cfg file: {dom_fn}')
+            return
+
+        e3t_0 = ds_dom.e3t_0
+
+        # Water column thickness, i.e. depth of bottom w-level on horizontal t-grid
+        H = e3t_0.cumsum(dim='z_dim').isel(z_dim=ds_dom.bottom_level.astype("int") - 1)
+        # Add correction to e3t_0 due to change in ssh
+        e3t_new = e3t_0 * (1 + ssh / H)
+        # preserve dimension ordering
+        e3t_new = e3t_new.transpose('t_dim', 'z_dim', 'y_dim', 'x_dim')
+        # mask out correction at layers below bottom level
+        e3t_new = e3t_new.where(e3t_new.z_dim < ds_dom.bottom_level, e3t_0.data)
+        # preserve any other t mask
+        e3t_new = e3t_new.where(~np.isnan(ssh))
+        if e3t:
+            e3_return.append(e3t_new.squeeze())
+
+        if np.any([e3u, e3v, e3f]):
```

Comment:
You could probably skip the list instantiation here and run any against the generator.